src/mpid/ch4/shm/posix/posix_init.c

```diff
@@ -235,14 +236,7 @@ int MPIDI_POSIX_coll_init(int rank, int size)
     }
     MPIR_ERR_CHECK(mpi_errno);
 
-    /* Actually allocate the segment and assign regions to the pointers */
-    mpi_errno = MPIDU_Init_shm_alloc(sizeof(int), &MPIDI_POSIX_global.shm_ptr);
-    MPIR_ERR_CHECK(mpi_errno);
-
-    MPIDI_POSIX_shm_limit_counter = (MPL_atomic_uint64_t *) MPIDI_POSIX_global.shm_ptr;
-
-    mpi_errno = MPIDU_Init_shm_barrier();
-    MPIR_ERR_CHECK(mpi_errno);
+    MPIDI_POSIX_shm_limit_counter = &MPIDI_POSIX_global_shm_ptr;
```

Comment:
This is incorrect. We need to allocate the `shm_ptr` on shared memory to make interprocess atomics work.